### PR TITLE
Story1

### PR DESCRIPTION
**Pull request by timyop**
This has been Unit tested, browser tested, accessibility tested and is responsive. hope it meets ur requirements and here if you have any questions

### CODE REVIEW

File: Style.css [hunk 1]
```diff
@@ -0,0 +1,56 @@
+
+html {
+
```
Comment (norayna) [inline Style.css]: empty line - delete

File: Style.css [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    box-sizing: border-box;
+    scroll-behavior: smooth;
+}
+
+body {
+    position: center;
+   width: 100vw;
```
Comment (norayna) [inline Style.css]: indent missing 1 space on this line

File: functions.php [hunk 3]
```diff
@@ -0,0 +1,42 @@
+<?php
+function dbpull():array{
```
Comment (norayna) [inline functions.php]: should be `dbPull()` - camelcase everywhere

File: functions.php [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    $connectionString = 'mysql:host=db; dbname=endangeredlist';
+    $dbUsername = 'root';
+    $dbPassword = 'password';
+    $db = new PDO($connectionString, $dbUsername, $dbPassword);
+    $db->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
+
+    $queryString = 'SELECT * FROM `animals`;';
+    $query = $db->prepare($queryString);
+    $query->execute();
+    $allresults = $query->fetchAll();
```
Comment (norayna) [inline functions.php]: should be `$allResults` - camelcase everywhere

File: functions.php [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    return $allresults;
+}
+$animals = dbpull();
```
Comment (norayna) [inline functions.php]: this line needs to be removed - it already runs in `index.php`

File: functions.php [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+function displayAnimals(array $animals):string {
```
Comment (norayna) [inline functions.php]: needs a space before `string`:
```php
function displayAnimals(array $animals): string {
```

File: functions.php [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+         $animalsHTML .=
+             '<div class = animal>' .
+                '<li class = species>Species: ' . $animal['species'] . '</li>'.
+                '<li class = facts>Individuals remaining :' . $animal['population'] . '</li>' .
+                '<li class = facts>Habitat: ' . $animal['country'] . '</li>' .
+                '<li class = facts>Charity: <a href = "' . $animal['charity'] . '">Link</a></li>' .
+            '</div>';
+     }
+     return $animalsHTML;
```
Comment (norayna) [inline functions.php]: indentation is weird in this function - i recommend doing Code > Reformat code in phpstorm to fix the indents in the whole file

File: functions.php [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+    $query = $db->prepare($queryString);
+    $query->execute();
+    $allresults = $query->fetchAll();
+    return $allresults;
+}
+$animals = dbpull();
+
+function displayAnimals(array $animals):string {
+    $animalsHTML = '';
+     foreach ($animals as $animal){
```
Comment (norayna) [inline functions.php]: need a space before the {

File: functions.php (hunk 3 above)
Comment (norayna) [reply functions.php]: also better to rename to `pullAllAnimalsFromDb()` or something like that - makes it a bit easier to tell exactly what it does

File: index.php [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <title>Collector App</title>
+    <link rel="stylesheet" type="text/css" href="normalize.css">
+    <link rel="stylesheet" type="text/css" href="Style.css">
+</head>
+<body>
+    <h1>ENDANGERED ANIMAL LIST</h1>
```
Comment (norayna) [inline index.php]: all caps should be done using css, the html code should be like this:
```html
<h1>Endangered Animals List</h1>
```

File: index.php [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+
+$animals = dbpull();
+?>
+
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <title>Collector App</title>
+    <link rel="stylesheet" type="text/css" href="normalize.css">
+    <link rel="stylesheet" type="text/css" href="Style.css">
```
Comment (norayna) [inline index.php]: your file `Style.css` should be renamed to `style.css`

File: index.php [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+</head>
+<body>
+    <h1>ENDANGERED ANIMAL LIST</h1>
+    <div>
+    <ul>
+        <?php
+        echo displayAnimals($animals);
```
Comment (norayna) [inline index.php]: this line should be indented 1 more tab inwards

File: test/functions.php [hunk 12]
```diff
@@ -0,0 +1,30 @@
+<?php
+require '../functions.php';
+
+use PHPUnit\Framework\TestCase;
+
+class functions extends TestCase
+{
+    public function testDisplayAnimalsSuccess()
```
Comment (norayna) [inline test/functions.php]: better name would be `testDisplayAnimalsSuccessString` and the other one `testDisplayAnimalsSuccessCorrectOutput`

File: test/functions.php [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+        $result = displayAnimals($animals);
+        $this->assertEquals($expected, $result);
+    }
+    public function testDisplayAnimalFailure()
+    {
+        $input = [['boat' => 'speed']];
+        $expected = '';
+        $result = displayAnimals($input);
+        $this->assertEquals($expected, $result);
+
```
Comment (norayna) [inline test/functions.php]: delete empty line

File: Style.css [hunk 14]
```diff
@@ -0,0 +1,57 @@
...
+    padding-top: 10px;
+}
+
+ul {
+    display: flex;
+    flex-direction: column;
+    align-content: center;
+    text-align: center;
+    margin-left: 0;
+    width: 100vw;
```
Comment (norayna) [inline Style.css]: better to use 100% rather than vw

File: index.php [hunk 15]
```diff
@@ -0,0 +1,25 @@
...
+        ?>
+    </ul>
+    </div>
```
Comment (norayna) [inline index.php:18]: this div can be removed (as discussed)

File: Style.css [hunk 16]
```diff
@@ -0,0 +1,57 @@
...
+html {
+
+    box-sizing: border-box;
+    scroll-behavior: smooth;
+}
+
+body {
+    position: center;
+   width: 100vw;
+    margin: 0 auto;
```
Comment (norayna) [inline Style.css]: what's the purpose of this line?

**Review by norayna — CHANGES_REQUESTED** [review]
upon reflection, i think you need to change up your html tags because `ul` and `li` are not really used in this way normally. here is my suggestion:

note that also the link is a different colour so that way it looks clickable. at the moment it's almost hidden
image

File: functions.php [hunk 17]
```diff
@@ -0,0 +1,36 @@
+<?php
+function pullAllAnimalsFromDb(): array
+{
+    $connectionString = 'mysql:host=db; dbname=endangeredlist';
+    $dbUsername = 'root';
+    $dbPassword = 'password';
+    $db = new PDO($connectionString, $dbUsername, $dbPassword);
+    $db->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
+
+    $queryString = 'SELECT * FROM `animals`;';
```
Comment (nealioa) [inline functions.php]: Replace the * with actual names of columns (in backticks).

An * is not recommended in production code.

File: functions.php [hunk 18]
```diff
@@ -0,0 +1,36 @@
...
+
+function displayAnimals(array $animals): string {
+    $animalsHTML = '';
+    foreach ($animals as $animal) {
+
+        if (!is_array($animal) ||
+            !array_key_exists('species', $animal) ||
+            !array_key_exists('population', $animal) ||
+            !array_key_exists('country', $animal) ||
+            !array_key_exists('charity', $animal)) {
```
Comment (nealioa) [inline functions.php]: Change this to
```
    !array_key_exists('charity', $animal)
) {
```

File: functions.php [hunk 19]
```diff
@@ -0,0 +1,36 @@
...
+        }
+
+        $animalsHTML .=
+            '<h2 class = species>Species: ' . $animal['species'] . '</h2>' .
+            '<p class = facts>Individuals remaining: ' . $animal['population'] . '</p>' .
+            '<p class = facts>Habitat: ' . $animal['country'] . '</p>' .
+            '<p class = facts>Charity: <a href = "' . $animal['charity'] . '">Link</a></p>';
+    }
+    return $animalsHTML;
+}
```
Comment (nealioa) [inline functions.php]: Add a blank line after this.

File: style.css [hunk 20]
```diff
@@ -0,0 +1,51 @@
...
+
+.animal {
+    display: flex;
+    flex-direction: column;
+    align-content: center;
+    text-align: center;
+    width: 100%;
+    list-style: none;
+    padding-left: 0;
+    border: 3px solid black;
```
Comment (nealioa) [inline style.css]: Replace the colour name with its hex value.

File: style.css [hunk 21]
```diff
@@ -0,0 +1,51 @@
...
+    font-weight: bolder;
+    font-size: 3rem;
+    color: #111111;
+    margin: 0;
+    padding-bottom: 5px;
+    border: 3px solid black;
+}
+
+.facts {
+    background-color: #85144b;
```
Comment (nealioa) [inline style.css]: To be consistent, hex values should all be uppercase or all lowercase.

**Review by nealioa — CHANGES_REQUESTED** [review]

**Review by nealioa — APPROVED** [review]

**Pull request merged**